manifests/base/runtimes/pretraining/mpi_distributed.yaml

```diff
@@ -12,36 +12,56 @@ spec:
       numProcPerNode: 1
       mpiImplementation: OpenMPI
       sshAuthMountPath: /home/mpiuser/.ssh
+      runLauncherAsNode: true
   template:
     spec:
       network:
-        publishNotReadyAddresses: false
+        publishNotReadyAddresses: true
       successPolicy:
         operator: All
         targetReplicatedJobs:
           - launcher
       replicatedJobs:
         - name: launcher
           template:
+            metadata:
+              labels:
+                trainer.kubeflow.org/trainjob-ancestor-step: trainer
```

Comment:
Please can we propagate this label all to way to ReplicatedJob Pods ?
```
rJob.spec.template.metadata.labels
```

We will use this label to detect appropriate pods in the SDK.

Reply:
Actually, sorry that is not required for now: https://github.com/kubeflow/trainer/blob/796645aee7c0be0aa0f4709eed0cac1fd1e38df5/sdk/kubeflow/trainer/api/trainer_client.py#L491-L499

We still need to rely on ReplicatedJob name to list Pods for now, since we need to get Launcher + Node pods for the MPI.